Update tspconfig.yaml

glharper opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Update tspconfig.yaml

glharper committed:
Merge branch 'dargilco/ai-model-inference' into glharper-inference-ts-dpg

glharper committed:
Update client.tsp

glharper committed:
Update client.tsp - remove embeddingsClient and ImageEmbeddingsClient

glharper committed:
Merge branch 'dargilco/ai-model-inference' into glharper-inference-ts-dpg

glharper committed:
Update tspconfig.yaml

glharper committed:
Update main.tsp

glharper committed:
Merge branch 'dargilco/ai-model-inference' into glharper-inference-ts-dpg

glharper committed:
Update main.tsp

glharper committed:
change namespace to Model

glharper committed:
use namespace AI.Model

glharper committed:
Use AI.Model instead of ModelClient

glharper committed:
Merge branch 'dargilco/ai-model-inference' into glharper-inference-ts-dpg

glharper committed:
Update client.tsp

glharper committed:
Delete specification/ai/ModelClient/models/image_embeddings.tsp

glharper committed:
Delete specification/ai/ModelClient/models/embeddings.tsp

glharper committed:
Update common.tsp

glharper committed:
Update routes.tsp

glharper committed:
Update client.tsp

glharper committed:
Update chat_completions.tsp

glharper committed:
Update chat_completions.tsp

glharper committed:
Merge branch 'dargilco/ai-model-inference' into glharper-inference-ts-dpg

glharper committed:
access fix

glharper committed:
use model keyword

glharper committed:
trying to fix ts gen

glharper committed:
more fixes

glharper committed:
more fixes

glharper committed:
revert chatcompletionsoptions to alias

glharper committed:
revert to alias

glharper committed:
Update tspconfig.yaml

glharper committed:
Update client.tsp

trangevi committed:
Merge branch 'main' into glharper-inference-ts-dpg

trangevi committed:
Attempt to merge Darren's branch (#32066)

* Go back to `parameters?: unknown` in FunctionDefinition model

* Add missing CompletionsFinishReason comments. Update tool choice related models to use the work `choice` instead of selection, similar to OAI TypeSpec

* Update JSON examples. Fix client.tsp after previous commit to rename models

* Add ID back to EmbeddingsResult model. Add comment to getModelInfo that it only works for MaaS/MaaP

* Update comments about hand-written operator methods

* Update teh ModelInfo enum

* Rename EmbeddingInput to ImageEmbeddingInput (#31953)

* Add Structured Outputs (#31962)

* Add support for structured outputs

Signed-off-by: trangevi <trangevi@microsoft.com>

* Naming consistency (#31966)

Signed-off-by: trangevi <trangevi@microsoft.com>

* Fixes following `npx tsp format **\*.tsp`

* Fix a couple of errors from `npx tsv .`

* Add missing class description. Add updated swagger file

* Update description

* Remove the in-line definition

* Make all response format models internal (#32005)

* Update client.tsp - alias ChatCompletionsResponseFormatJsonObject in JS to avoid breaking change

* Update chat_completions.tsp

---------

Signed-off-by: trangevi <trangevi@microsoft.com>
Co-authored-by: Darren Cohen <39422044+dargilco@users.noreply.github.com>
Co-authored-by: Glenn Harper <64209257+glharper@users.noreply.github.com>

glharper committed:
Make ChatCompletionsOptions public in Java

glharper committed:
Merge remote-tracking branch 'origin/dargilco/ai-model-inference' into glharper-inference-ts-dpg

glharper committed:
add alias for ChatCompletionsResponseFormatJsonObject

glharper committed:
add developer chat role

glharper committed:
remove java specific augment

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[AI.Model](https://apiview.dev/Assemblies/Review/aa230d385fde45d69086ce67b31b9f48?revisionId=44d6f93819314ec3aefcaf7ddcebca99)

glharper committed:
use correct names for audio content classes

glharper committed:
fix tsp error

raych1 committed:
New workflow for sdk breaking change labeling (#33158)

* Added WF for sdk breaking change labeling

* publish label artifact separately

* use step with condition

* use customcondition for publish template

* added tests

* use check run from payload

* refactored logging and process label artifact

* read spec-gen-sdk report to create label artifact

* complete run without op if artifact not found

* Updated package.json

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

yuxiz1993 committed:
change retry-after to string in 2025-03-01 examples (#33304)

Co-authored-by: Yuxi Zhang <yuxzha@microsoft.com>

raych1 committed:
Updated tests to ignore logging verfication (#33363)

welovej committed:
rename-instanceCountsByState-for-breaking-change (#33376)

dependabot[bot] committed:
Bump the typespec group across 1 directory with 2 updates (#33371)

Bumps the typespec group with 2 updates in the / directory: [@azure-tools/typespec-apiview](https://github.com/Azure/azure-sdk-tools) and [@azure-tools/typespec-client-generator-core](https://github.com/Azure/typespec-azure).


Updates `@azure-tools/typespec-apiview` from 0.5.1 to 0.6.0
- [Release notes](https://github.com/Azure/azure-sdk-tools/releases)
- [Commits](https://github.com/Azure/azure-sdk-tools/compare/doc-warden_0.5.1...doc-warden_0.6.0)

Updates `@azure-tools/typespec-client-generator-core` from 0.53.0 to 0.53.1
- [Release notes](https://github.com/Azure/typespec-azure/releases)
- [Commits](https://github.com/Azure/typespec-azure/compare/typespec-azure@0.53.0...@azure-tools/typespec-client-generator-core@0.53.1)

---
updated-dependencies:
- dependency-name: "@azure-tools/typespec-apiview"
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: typespec
- dependency-name: "@azure-tools/typespec-client-generator-core"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: typespec
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mikeharder committed:
[dependabot.yml] Update for eng/tools/lint-diff (#33383)

mikeharder committed:
[dependabot.yml] Ignore "js-yaml" (#33385)

adyabansal-ms committed:
Updating type of endorsement from string to array of string (#33310)

* Updating type of endorsement from string to array of string

* Adding typespec suppression for catalog

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

mikeharder committed:
[.github] Add e2e tests for git helpers (#33387)

- Fixes #33334
- Detect issues missed by mock tests

alluri02 committed:
Update Public Repo with Liftr Neon Preview Versions from Private Repo (#33311)

* update public repo with preview versions from the private repo

* update public repo with preview version from the private repo

* update readme with preview versions

* updated examples in the swagger json

* fix swagger model validations

* fix .Net error, changed the name of response model to result

* added 2 new attributes for project

* fix swagger model validation and prettier errors

---------

Co-authored-by: Srinivas Alluri <sralluri@microsoft.com>

azure-sdk committed:
Package Properties processing updates (#33396)

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

jpalvarezl committed:
Made usage optional in ChatCompletions and Completions (#33315)

mikeharder committed:
[.github] Verify all modules are importable (#33390)

stevemunk committed:
Corrected capitalization for waypoint. (#33201)

Co-authored-by: Mike Harder <mharder@microsoft.com>

azure-sdk committed:
Removed obsolete specc-gen-sdk pipeline template (#33397)

Co-authored-by: ray chen <raychen@microsoft.com>

skapur12 committed:
Turn off generated samples and tests for Java (#33404)

dependabot[bot] committed:
Bump @microsoft.azure/openapi-validator from 2.2.3 to 2.2.4 in /eng/tools/lint-diff (#33374)

* Bump @microsoft.azure/openapi-validator in /eng/tools/lint-diff

Bumps [@microsoft.azure/openapi-validator](https://github.com/Azure/azure-openapi-validator) from 2.2.3 to 2.2.4.
- [Changelog](https://github.com/Azure/azure-openapi-validator/blob/main/changelog.md)
- [Commits](https://github.com/Azure/azure-openapi-validator/commits)

---
updated-dependencies:
- dependency-name: "@microsoft.azure/openapi-validator"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* npm i

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Mike Harder <mharder@microsoft.com>

rkuthala committed:
Remove unsupported filter for cognitive service list usages API (#33108)

Co-authored-by: Ramesh Kuthala <rkuthala@microsoft.com>

raych1 committed:
Use original file name when searching files (#33402)

* use original file name when search files

* Log warning issues in pipeline result

danieljurek committed:
Skip test that sometimes fails with mock.calls.length of 2 instead of expected 1 (#33403)

* Skip test that sometimes fails with mock.calls.length of 2 instead of expected 1

* Proper syntax

* Use test.sequential for getRelatedArmRpcFromDoc which could have race conditions

danieljurek committed:
More obvious testing naming and logging (#33401)

* More obvious testing naming and logging

* Revert changes to lint-diff.ts

* Echo ignore message after running lint-diff

* echo -e

* simulate failure (TODO REVERT)

* always

* Prepend behavior

* .

* Revert failure simulation

* Apply suggestions from code review

Co-authored-by: Mike Harder <mharder@microsoft.com>

* Revert changes to LintDiff test workflow

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

danieljurek committed:
Revert workflow name to Swagger LintDiff - Test (#33406)

* Revert workflow name

* Quotes

noahdavis-msft committed:
Release azuredatatransfer microsoft.azure data transfer preview/2025 03 01 preview (#33212)

* Copy files from stable/2024-09-27

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2025-03-01-preview

Updated the API version from stable/2024-09-27 to preview/2025-03-01-preview.

* Added tag for 2025-03-01-preview in readme file

* Add changes from PR

dandandanliu915 committed:
[SnowflakeV2] Updated linked service (#33270)

* [Snowflake] Updated linked service

* Apply the same in synapse

* remove

---------

Co-authored-by: Dan Liu <liudan7@microsoft.com>
Co-authored-by: Jingshu Pan <v-jinpan@microsoft.com>

mikeharder committed:
[ARM Auto Signoff] Use latest run of ARM Incremental TypeSpec (#33412)

- Fixes #32991
- Adds unit tests for 100% coverage

mikeharder committed:
[.github] Add tests for 100% coverage (#33414)

mikeharder committed:
[context.js] Fallback to search API to find PR from head_sha (#33420)

- Fixes #33415
- Fixes #33416
- Fixes #33417

pshao25 committed:
[HybridConnectivity] Replace ArmResourcePatch to ArmCustomPatch (#33275)

* Replace ArmResourcePatch to ArmCustomPatch

* update

* update

---------

Co-authored-by: melina5656 <140602263+melina5656@users.noreply.github.com>
Co-authored-by: Judy Liu <v-liujudy@microsoft.com>

JiaqiZhang-Dev committed:
fix go-emitter options (#32937)

TheDonLuke committed:
Adding the FIST 2025-04-01-preview API  (#33298)

* adding 2025-04-01-preview api

* Fixing doc string for accuracy

* missed copying the cspell file from the PR repo

---------

Co-authored-by: Mike Lucadamo <mlucadamo@microsoft.com>

Frey-Wang committed:
[ADF] Add support for M365 SP Cert authentication (#33176)

* [ADF] Add support for M365 SP Cert authentication

* update for synapse

dependabot[bot] committed:
Bump @eslint/js from 9.22.0 to 9.23.0 in /.github (#33432)

Bumps [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js) from 9.22.0 to 9.23.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/commits/v9.23.0/packages/js)

---
updated-dependencies:
- dependency-name: "@eslint/js"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 9.22.0 to 9.23.0 in /.github (#33431)

Bumps [eslint](https://github.com/eslint/eslint) from 9.22.0 to 9.23.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.22.0...v9.23.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mikeharder committed:
[dependabot.yml] Add group for eslint (#33438)

v-jiaodi committed:
[1] Update tspconfig options with kebab case style (#33338)

* update tspconfig options with kebab case style

* revert error case

---------

Co-authored-by: Mary Gao <yanmeigao1210@gmail.com>

dependabot[bot] committed:
Bump @tsconfig/node20 from 20.1.4 to 20.1.5 in /.github (#33439)

Bumps [@tsconfig/node20](https://github.com/tsconfig/bases/tree/HEAD/bases) from 20.1.4 to 20.1.5.
- [Commits](https://github.com/tsconfig/bases/commits/HEAD/bases)

---
updated-dependencies:
- dependency-name: "@tsconfig/node20"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

stevemunk committed:
removed 'important' banner about using the preview version. (#33407)

stevemunk committed:
Added a retirement banner to Azure Maps Route v1 APIs (#33405)

* Added retirement banner

* Added retirement banner

v-jiaodi committed:
[7] Update tspconfig options with kebab case style (#33348)

* update tspconfig options with kebab case style

* revert error case

---------

Co-authored-by: Mary Gao <yanmeigao1210@gmail.com>

v-jiaodi committed:
[2] Update tspconfig options with kebab case style (#33343)

* update tspconfig options with kebab case style

* revert error case

* revert error case

* revert error case

---------

Co-authored-by: Mary Gao <yanmeigao1210@gmail.com>

v-jiaodi committed:
[3] Update tspconfig options with kebab case style (#33344)

* update tspconfig options with kebab case style

* revert error case

* revert error case

---------

Co-authored-by: Mary Gao <yanmeigao1210@gmail.com>

v-jiaodi committed:
[4] Update tspconfig options with kebab case style (#33345)

* update tspconfig options with kebab case style

* revert error case

* revert error case

v-jiaodi committed:
[5] Update tspconfig options with kebab case style (#33346)

* update tspconfig options with kebab case style

* revert error case

* revert error case

v-jiaodi committed:
[6] Update tspconfig options with kebab case style (#33347)

* update tspconfig options with kebab case style

* revert error case

* revert error case

---------

Co-authored-by: Mary Gao <yanmeigao1210@gmail.com>

mitsha-microsoft committed:
[AzureLoadTesting] Renames for Models for better SDK Generation (#33443)

Renamed enums and removed model generation for .NET

trangevi committed:
Merge branch 'main' into glharper-inference-ts-dpg

csurf987 committed:
New release version stable/2025-03-01 for Microsoft.CostManagement (#33189)

* Copy files from stable/2024-08-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2025-03-01

Updated the API version from stable/2024-08-01 to stable/2025-03-01.

* Added tag for 2025-03-01 in readme file

* Update exports json to support the new behaviors included in the 2023-07-01-preview API version.

* Updated examples and documentation for latest updates

* Added new examples, updated destination type and nextRunTimeEstimate in existing examples

* Fix spell check errors

* Swagger PrettierCheck

* Fix swagger model validation

* Updates to readme, fix casing in compressionMode to be consistent

* Revert changes from readme.go.md

* Revert "Revert changes from readme.go.md"

This reverts commit 5ccd07f6a5f4d1c5b24a92fddb814e987af31ac6.

* Revert "Updates to readme, fix casing in compressionMode to be consistent"

This reverts commit 8b8d15b5a5f7d143883f07a40ed15f87ee0e0d47.

* Update tag and enum value casing

* Update readme.go.md

* Revert "Update readme.go.md"

This reverts commit c2236d9fc99d0638115186c39476e996a5f2b4ce.

* Update readme.go.md

* Remove 'Delta' from export format options as per review comments

* Update tag in Cost Management API readme

Fixing format tag

* Create suppressions.yaml

---------

Co-authored-by: Meera Shanmugavel <meshanmu@microsoft.com>

mikeharder committed:
[ARM Auto SignOff] Update for GA (#33440)

- Remove "preview" from all files and strings
- Add label "ARMSignedOff" in addition to "ARMAutoSignOff"
- Only add "issue-number" artifact if adding or removing label (perf)

amber-ccc committed:
fix the naming issues for authoring (#33259)

* fix the name issue in authoring and generate new swaggers

* adjust the tests for Conversation Authoring

* adjusted the tests for text authoring

* fixed Typespec validation

mikeharder committed:
[ARM Auto SignOff] Only remove ARMSignedOff if ARMAutoSignedOff (#33461)

- Necessary to prevent removing "ARMSignedOff" if added by a human reviewer

ChenxiJiang333 committed:
[Dependency Map] Update tspconfig.yaml (#33436)

* Update tspconfig.yaml

* update tsp config

najian committed:
Release app microsoft.app 2025 01 01 (#32578)

* Copy files from stable/2024-03-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2025-01-01

Updated the API version from stable/2024-03-01 to stable/2025-01-01.

* Added tag for 2025-01-01 in readme file

* Support keda scaler auth using MSI (#31893)

* Support keda scaler auth using MSI

* fix lint

* fix lint

* update

* Add KEDA cooldownPeriod/pollingInterval property for Container App & Jobs (#31891)

* add keda propety

* fix prettier

* fix comment

* Enable BYO NFS Storage in 2025-01-01 (#31850)

* Enable BYO NFS Storage in 2025-01-01

* Fix code style

* Fix

---------

Co-authored-by: Layla Liu <layliu@microsoft.com>

* Add RunningStatus for ContainerApp (#31984)

* Enable Java Runtime in Microsoft.App 2025-01-01 (#31957)

* Enable Java Runtime in 2025-01-01
---------

Co-authored-by: najian <jiangnan8924@gmail.com>

* Support env msi and cert from key vault (#31875)

* Support env msi and cert from key vault

* add spell check ci

* Support Java Components in Microsoft.App version 2025-01-01 (#31958)

* add java components support in version 2025-01-01

* enable lenient model deduplication in go

---------

Co-authored-by: Chenjie Shi <tadelesh.shi@live.cn>

* Update proxy resource type to v5 (#31997)

* update proxy resource type to v5

---------

Co-authored-by: Nan Jiang <naji@microsoft.com>

* remove deduplication config for go sdk (#32111)

* Azure Container Apps SessionPool Api Version 2025-01-01 (#31985)

* SessionPool Api Version 2025-01-01

* Fix example

* DynamicPoolConfiguration.LifecycleConfiguration

* Fix

* Test

* Test

* Test

* test

* Test

* Test

* Test

* Test

* Test

* Test

* Restore

* Test

* Test

* Test

* Restore

* Use TrackedResource for SessionPool patch

* Remove tracked resource and only support tags for patch

* Fix lintdiff

* Update lifecycle type

* Add supress

* fix

* Fix LintDiff failed in Release app microsoft.app 2025 01 01 (#33078)

* Fix LintDiff failed
---------

Co-authored-by: Nan Jiang <naji@microsoft.com>

* Adding managed identity settings (#32009)

* Don't add 201 according to breaking change review comments (#33094)

* Revert "Fix LintDiff failed in Release app microsoft.app 2025 01 01 (#33078)"

This reverts commit 58c987f6f83a959b7dda7bb27bb7e5d651c05894.

* Fix example

* suppression PutResponseCodes per Breaking change review comments

---------

Co-authored-by: Nan Jiang <naji@microsoft.com>

---------

Co-authored-by: Nan Jiang <naji@microsoft.com>
Co-authored-by: njucz <740360112@qq.com>
Co-authored-by: Zunli Hu <zuh@microsoft.com>
Co-authored-by: LaylaLiu-gmail <38268900+LaylaLiu-gmail@users.noreply.github.com>
Co-authored-by: Layla Liu <layliu@microsoft.com>
Co-authored-by: zhenqxuMSFT <zhenqxu@microsoft.com>
Co-authored-by: Songbo Wang <sonwan@microsoft.com>
Co-authored-by: Chenjie Shi <tadelesh.shi@live.cn>
Co-authored-by: Yitao Pan <yitaopan@microsoft.com>
Co-authored-by: Mike Harder <mharder@microsoft.com>
Co-authored-by: Vaclav Turecek <vturecek@microsoft.com>

ChenxiJiang333 committed:
Update readme.python.md (#33434)

alluri02 committed:
added stable version for Neon GA (#33437)

* added stable version for Neon GA

* fix email regex issue

* fix the arm id invalid format issue

* fix typespec validation

---------

Co-authored-by: Srinivas Alluri <sralluri@microsoft.com>
Co-authored-by: Mike Harder <mharder@microsoft.com>

dependabot[bot] committed:
Bump @typespec/compiler in the typespec group across 1 directory (#33467)

Bumps the typespec group with 1 update in the / directory: [@typespec/compiler](https://github.com/microsoft/typespec).


Updates `@typespec/compiler` from 0.67.1 to 0.67.2
- [Release notes](https://github.com/microsoft/typespec/releases)
- [Commits](https://github.com/microsoft/typespec/compare/@typespec/compiler@0.67.1...@typespec/compiler@0.67.2)

---
updated-dependencies:
- dependency-name: "@typespec/compiler"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: typespec
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @azure-tools/spec-gen-sdk from 0.3.0 to 0.3.1 (#33468)

Bumps [@azure-tools/spec-gen-sdk](https://github.com/Azure/azure-sdk-tools) from 0.3.0 to 0.3.1.
- [Release notes](https://github.com/Azure/azure-sdk-tools/releases)
- [Commits](https://github.com/Azure/azure-sdk-tools/compare/doc-warden_0.3.0...doc-warden_0.3.1)

---
updated-dependencies:
- dependency-name: "@azure-tools/spec-gen-sdk"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint-plugin-unicorn from 57.0.0 to 58.0.0 in /eng/tools/specs-model in the eslint group across 1 directory (#33469)

* Bump eslint-plugin-unicorn

Bumps the eslint group with 1 update in the /eng/tools/specs-model directory: [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn).


Updates `eslint-plugin-unicorn` from 57.0.0 to 58.0.0
- [Release notes](https://github.com/sindresorhus/eslint-plugin-unicorn/releases)
- [Commits](https://github.com/sindresorhus/eslint-plugin-unicorn/compare/v57.0.0...v58.0.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-unicorn
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: eslint
...

Signed-off-by: dependabot[bot] <support@github.com>

* npm i

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Mike Harder <mharder@microsoft.com>

dishantmunjal committed:
Adding Identity Details in Examples of Backup Instance Operations (#33272)

* Adding Identity Details in Examples of Backup Instance Operations

mikeharder committed:
[chore] npm update (#33474)

robga committed:
Robga/exemption228 (#32923)

* copied from the old version

* Add 2024-12-01-preview for exemptions

* missed to stage the change

* Fix linter

* Fix common type file version

* Fixed typo

viveksheelsingh committed:
Changes for new API Version 2024-11-01-preview... (#33389)

* Copy files from stable/2024-10-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-11-01-preview

Updated the API version from stable/2024-10-01 to preview/2024-11-01-preview.

* Added tag for 2024-11-01-preview in readme file

* adding delta to base

* adding examples

* fixing prettier

* fixing api version

* fixing ackage tag for latest version

* fixing package tag for latest version

* fixing example

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

koderjoker committed:
Release redisenterprise microsoft.cache 2025 04 01 (#33123)

* Copy files from preview/2024-09-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2025-04-01

Updated the API version from preview/2024-09-01-preview to stable/2025-04-01.

* Added tag for 2025-04-01 in readme file

* Add missing changes from stable 2024-10-01 to stable 2025-04-01 based off preview 2024-09-01

* Add new 'Moving' ResourceState to enum

* Cleanup by prettier

---------

Co-authored-by: Kanchan Joshi <kanchanjoshi@microsoft.com>
Co-authored-by: Mike Harder <mharder@microsoft.com>

ryanmaley310 committed:
Adding format to DiscountPercentage definition (#33456)

raych1 committed:
Set pipeline variables used to create SDK PR and refactored loading execution report (#33462)

* set pipeline variables in runner and refactored get execution report

* Removed condition of publish log step

* revert lint config

* Upgrade spec-gen-sdk version

minhsuanlee committed:
Update OperationResults to be location-based in 2024-09-01 (#33035)

* Updated swagger and examples

* Add spec for Azure Site Recovery service

* More fixes

* Did more fixes

* Update OperationResults to be location-based in 2024-09-01

* Remove files added by accident

* Update client config for .Net code generation

* update

* update

* update

* update

---------

Co-authored-by: Sam Lee <samlee3@microsoft.com>
Co-authored-by: Arthur Ma <arthurma@microsoft.com>
Co-authored-by: Roopesh Manda <rmanda@microsoft.com>

csurf987 committed:
Revert "New release version stable/2025-03-01 for Microsoft.CostManagement" (#33481)

* Revert "New release version stable/2025-03-01 for Microsoft.CostManagement (#…"

This reverts commit 1880675cfec07301a861b2cb72a94ff0282affe8.

* Create suppressions.yaml

yuxiz1993 committed:
Microsoft.quota create 2025-03-15-preview version (#33228)

* Add new preview version

* Add groupType and location

* Add readme

* merge with the stable version

* merge example

* change retry-after to string

* Revert "change retry-after to string"

This reverts commit 43af49b77f6b37729ca0528d075cf5b9e641a5e5.

* Reapply "change retry-after to string"

This reverts commit c3b795f2064a79cf52f1a081e755e5a872160fc7.

* Change retry-after to string

* Revert "Change retry-after to string"

This reverts commit 377a48f1922818df3706d523117c2bc17db2e95e.

* Change retry-after type to string in 2024-12-18-preview

* change the retry-after type back to int in spec

* Change retry-after to string in 2025-03-15-preview examples

* minor fix

* Change readme.ruby default api ver

* Pump python ver

---------

Co-authored-by: Yuxi Zhang <yuxzha@microsoft.com>

curiekim00 committed:
Users/curiekim/add 2025 03 01 preview databricks (#32478)

* Copy files from preview/2024-09-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2025-03-01-preview

Updated the API version from preview/2024-09-01-preview to preview/2025-03-01-preview.

* Added tag for 2025-03-01-preview in readme file

* update README with the new API version and databricks.json file with the new enum details

* Fix spellcheck and prettiercheck issues

* fix prettiecheck issues

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 10129 (#33486)

* identify and fix the issue with no results on a matrix that is PRBatched. The platform filters eliminate all the jobs, which is why it only turned up on the second windows leg

---------

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

azure-sdk committed:
Add option to create self contained test resources post script (#33366)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

IannGeorges committed:
[Az.DataProtection] New ADLS backup parameter (#32616)

* Copy files from stable/2025-01-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2025-02-01

Updated the API version from stable/2025-01-01 to stable/2025-02-01.

* Added tag for 2025-02-01 in readme file

* initial commit - adding new adls backup parameter + examples

* prettier changes

* additional prettier changes

* add ref to adls example

* fix model validation & prettier checks

* override spell check for adls

* Remove allOf inheritted properties from swagger json

* fix lint issues

---------

Co-authored-by: ianna1-admin <ianna@microsoft.com>
Co-authored-by: Arthur Ma <mars.arthur@263.net>

mikeharder committed:
[.github] Add dep for json schema ref parsing (#33494)

- Fixes #33475
- Replace "yaml" with "js-yaml", since the latter is a dep of "@apidevtools/json-schema-ref-parser"

iscai-msft committed:
[python] switch package-name -> namespace (#32282)

* switch package-name -> namespace

* add namespace

* remove unnecessary namespace config

---------

Co-authored-by: iscai-msft <isabellavcai@gmail.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>
Co-authored-by: ChenxiJiang333 <v-chenjiang@microsoft.com>
Co-authored-by: Renhe Li <renhe.li@hotmail.com>

kazrael2119 committed:
Try to remove hierchyClient configs for typespec-ts in tspconfig (#33210)

* Try to remove hierchyClient configs for typespec-ts in tspconfig

* update

* update

* update

---------

Co-authored-by: Qiaoqiao Zhang <55688292+qiaozha@users.noreply.github.com>
Co-authored-by: Wanpeng Li <wanl@microsoft.com>

stevemunk committed:
Added retirement banner (#33450)

adityareddy305 committed:
change empty obj to record<unknown>, sdk regenartion works better thi… (#33352)

* change empty obj to record<unknown>, sdk regenartion works better this way for accepting dynamic properties

* add supressions

* add where caluse for supressions

trangevi committed:
fix anonymous object error (#33514)

Signed-off-by: trangevi <trangevi@microsoft.com>

Commit:
Fix enum divergence (#33422)

* Update computeRPCommon.json

* update x-ms-enum name

* Update computeRPCommon.json

* Rename `HostEndPointSettingMode` to `Modes`

* Update virtualMachineScaleSet.json

mikeharder committed:
[github] Add "cwd" (current working directory) to options bags (#33520)

- Keep defaulting to GITHUB_WORKSPACE if available, else process.cwd()
- Fixes #33284

azure-sdk committed:
Import Az powershell module in self contained post script task (#33521)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

dependabot[bot] committed:
Bump @azure-tools/typespec-client-generator-cli (#33491)

Bumps the typespec-client-generator-cli group with 1 update in the / directory: [@azure-tools/typespec-client-generator-cli](https://github.com/Azure/azure-sdk-tools).


Updates `@azure-tools/typespec-client-generator-cli` from 0.15.3 to 0.16.0
- [Release notes](https://github.com/Azure/azure-sdk-tools/releases)
- [Commits](https://github.com/Azure/azure-sdk-tools/commits)

---
updated-dependencies:
- dependency-name: "@azure-tools/typespec-client-generator-cli"
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: typespec-client-generator-cli
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

koyasu221b committed:
Update the Route document to improve the appearance of the overview page (#33502)

* update

* Add space

---------

Co-authored-by: Will Huang <huangwill@microsoft.com>

raych1 committed:
Support new parameters in pipeline (#33488)

* Added sdk-release-type and api-version to pipeline

* fixed syntax error

* Removed specRepoCommit parameter

* Removed unnecessary commit validation step

* return fail status code

* skip all jobs for unofficial template

* fixed syntax errors

* add pool

* set flag in sdl option

* test with unofficial only

* fix template name

* revert template change

* prefix the name with 'test-ignore'

* Use preview in job name

* remove default value

* explicitly add pullrequest constraint to condition for better readability

* revert change of default value removal as it's needed for spec PR scenario

mikeharder committed:
[.github] Move GITHUB_WORKSPACE into workflows folder (#33528)

trangevi committed:
Merge main inference branch into js feature branch (#33451)

* Apply new local cspell

* Fix wrongly deleted new cspell in the root

* Add developer role to support new GPT models (#32474)

* Fix tcgc error due to Record<unknown> (#32719)

* Fix tcgc error due to Record<unknown>

Signed-off-by: trangevi <trangevi@microsoft.com>

* Update other options objects to model

Signed-off-by: trangevi <trangevi@microsoft.com>

* Fix compile errors. Update swagger

Signed-off-by: trangevi <trangevi@microsoft.com>

---------

Signed-off-by: trangevi <trangevi@microsoft.com>

* Merge C# feature branch into primary Azure.AI.Inference branch (#31869)

* Update ImageEmbeddingsClient to change embed to embedImage

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Add typespec-csharp to the tspconfig

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Move csharp client property renames

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Missing semicolons

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Wrong namespace

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Incorrect namespaces

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Aliases need to have things referenced differently, apparently

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Need to specify parameters?

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Move alias renames to the individual file for now

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Change ChatCompletionsOptions to a model

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* syntax

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Leave the model property

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Add additional properties for ChatCompletionsOptions

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* 3 dots, not 2

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Can't add docstring to generic spread property

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Change unknown parameters to extra parameters

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Remove unnecessary rename

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Removing api-key change for python, for now

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Make operator method internal

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Fix pass-through header

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Re-enable protocol methods for csharp

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Remove "spread" for ChatCompletionsOptions

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Remove spread for additionalRequestHeaders

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* needed commas

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* another formatting attempt

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* okay, figured out how the formatting works

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Alias breaks the generator

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* A model won't give us the interaction pattern that we want. Trying this.

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Forgot semicolon

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Does swapping the order of parameters matter?

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Make extraParams optional?

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Try and get rid of anonymous model

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Try adding RequestHeadersTrait

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Maybe needs to be an object?

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Add using

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Revert to "good" state

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Try moving extra parameters. Change Parameters back to unknown

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Skip the alias

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Is it the header which is causing issues?

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Headers in the options object seems to not work

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Remove commented operation

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Unnecessary using

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Add embeddings for csharp

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Client changes for C#

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Need the options objects to be models

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Make the embedding methods public for now, for intial testing

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Remove image embedding for now until it can be tested

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Add new streaming classes for C#

Signed-off-by: Travis Angevine <trangevi@microsoft.com>

* Revert to Darren's branch to test

Signed-off-by: trangevi <trangevi@microsoft.com>

* Explicitly set the completions options object to be an input

Signed-off-by: trangevi <trangevi@microsoft.com>

* Other clients with spread parameters

Signed-off-by: trangevi <trangevi@microsoft.com>

* No image embedding client yet, so leave this out for now

Signed-off-by: trangevi <trangevi@microsoft.com>

* Operations need to be internal for all languages

Signed-off-by: trangevi <trangevi@microsoft.com>

* Comment

Signed-off-by: trangevi <trangevi@microsoft.com>

* Add support for structured outputs

Signed-off-by: trangevi <trangevi@microsoft.com>

* We already had the property, just need to add one more option

Signed-off-by: trangevi <trangevi@microsoft.com>

* Remove unnecessary inner class

Signed-off-by: trangevi <trangevi@microsoft.com>

* Rename

Signed-off-by: trangevi <trangevi@microsoft.com>

* extra semicolon

Signed-off-by: trangevi <trangevi@microsoft.com>

* Fix naming

Signed-off-by: trangevi <trangevi@microsoft.com>

* What if it's just unknown?

Signed-off-by: trangevi <trangevi@microsoft.com>

* unknown to record unknown

Signed-off-by: trangevi <trangevi@microsoft.com>

* Leave as unknown for now

Signed-off-by: trangevi <trangevi@microsoft.com>

* Add image embeddings client back to csharp

Signed-off-by: trangevi <trangevi@microsoft.com>

* Generation is fixed, switch back to record

Signed-off-by: trangevi <trangevi@microsoft.com>

* Make ImageEmbeddingsOptions explicitly an input

Signed-off-by: trangevi <trangevi@microsoft.com>

* First attempt. Make the main content item a parent of the two relevant ones, will require a custom factory

Signed-off-by: trangevi <trangevi@microsoft.com>

* Do I need this explicitly public?

Signed-off-by: trangevi <trangevi@microsoft.com>

* What about usage?

Signed-off-by: trangevi <trangevi@microsoft.com>

* More inputs

Signed-off-by: trangevi <trangevi@microsoft.com>

* Remove middle layer

Signed-off-by: trangevi <trangevi@microsoft.com>

* Missed some

Signed-off-by: trangevi <trangevi@microsoft.com>

* TSP formatting and swagger changes

Signed-off-by: trangevi <trangevi@microsoft.com>

* Another format change

Signed-off-by: trangevi <trangevi@microsoft.com>

* Update with some changes to better reflect the desired merge

Signed-off-by: trangevi <trangevi@microsoft.com>

* Mark options object as internal to address transitive access issue

Signed-off-by: trangevi <trangevi@microsoft.com>

* review comments

Signed-off-by: trangevi <trangevi@microsoft.com>

* Add renaming for python classes

Signed-off-by: trangevi <trangevi@microsoft.com>

* Update after npx command

Signed-off-by: trangevi <trangevi@microsoft.com>

---------

Signed-off-by: Travis Angevine <trangevi@microsoft.com>
Signed-off-by: trangevi <trangevi@microsoft.com>

---------

Signed-off-by: trangevi <trangevi@microsoft.com>
Signed-off-by: Travis Angevine <trangevi@microsoft.com>
Co-authored-by: Darren Cohen <39422044+dargilco@users.noreply.github.com>

trangevi committed:
Merge branch 'main' into glharper-inference-ts-dpg

trangevi commented:
Closed with permission from Glenn, will reopen to clean up review list
